### PR TITLE
Added writing of `genesis.Alloc` on bootstrap

### PR DESCRIPTION
**Pull request by maoueh**

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -406,6 +406,34 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 		}
 	}
 
+	if bc.logger != nil && bc.CurrentBlock().Number.Uint64() == 0 {
```
Comment (s1na) [inline core/blockchain.go]: Hm, so we want to only emit this event on the first sync? Asking because you have the `currentBlock == 0` check

Comment (maoueh) [reply core/blockchain.go]: I'm not sure we have the state that tells us if it's a first sync or not. I would like to have `OnGenesisBlock` called only once indeed!

**Comment by maoueh** [comment]
Updated with the fact that genesis state spec is already written, only modified `geth init` now to also write the state spec if not present in the database

File: core/genesis.go [hunk 2]
```diff
@@ -167,45 +167,68 @@ func (ga *GenesisAlloc) flush(db ethdb.Database, triedb *trie.Database, blockhas
...
-		case params.GoerliGenesisHash:
-			genesis = DefaultGoerliGenesisBlock()
-		case params.SepoliaGenesisHash:
-			genesis = DefaultSepoliaGenesisBlock()
-		}
-		if genesis != nil {
-			alloc = genesis.Alloc
-		} else {
-			return errors.New("not found")
+			return nil, true, err
```
Comment (s1na) [inline core/genesis.go]: Do we really need `found`? a `nil` alloc could mean not found

File: core/genesis.go [hunk 3]
```diff
@@ -167,45 +167,68 @@ func (ga *GenesisAlloc) flush(db ethdb.Database, triedb *trie.Database, blockhas
...
+// CommitGenesisState loads the stored genesis state with the given block
+// hash and commits it into the provided trie database.
+func CommitGenesisState(db ethdb.Database, triedb *trie.Database, blockhash common.Hash) error {
+	alloc, found, err := getGenesisState(db, blockhash)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		return nil
```
Comment (s1na) [inline core/genesis.go]: We used to return `errors.New("not found")` in this case. What's the reason for this behavior change?

File: core/genesis.go (hunk 2 above)
Comment (maoueh) [reply core/genesis.go]: Yes we can do that.

File: core/genesis.go (hunk 3 above)
Comment (maoueh) [reply core/genesis.go]: That's was a mistake indeed, I'll revert this

**Review by s1na — APPROVED** [review]
LGTM

**Pull request merged**